spec/automation/unit/method_validation/check_pre_retirement_spec.rb

```diff
@@ -1,34 +1,71 @@
 describe "check_pre_retirement Method Validation" do
-  before(:each) { @vm = FactoryGirl.create(:vm_microsoft) }
-  after(:each) { @vm.destroy }
+  let(:user) { FactoryGirl.create(:user_with_group) }
+  let(:zone) { FactoryGirl.create(:zone) }
 
-  let(:user) do
-    FactoryGirl.create(:user_with_group)
-  end
+  context "Infrastructure" do
+    let(:ems) { FactoryGirl.create(:ems_microsoft, :zone => zone) }
+    let(:vm) do
+      FactoryGirl.create(:vm_microsoft,
+                         :raw_power_state => "PowerOff",
+                         :ems_id          => ems.id)
+    end
+    let(:ws) do
+      MiqAeEngine.instantiate("/Infrastructure/VM/Retirement/StateMachines/Methods/CheckPreRetirement?" \
+                              "Vm::vm=#{vm.id}#microsoft", user)
+    end
 
-  let(:ws) do
-    MiqAeEngine.instantiate("/Infrastructure/VM/Retirement/StateMachines/Methods/CheckPreRetirement?" \
-                            "Vm::vm=#{@vm.id}#microsoft", user)
-  end
+    it "returns 'ok' for a vm in powered_off state" do
+      expect(ws.root['vm'].power_state).to eq("off")
+      expect(ws.root['ae_result']).to eq("ok")
+    end
 
-  it "returns 'ok' for a vm in powered_off state" do
-    @vm.update_attribute(:raw_power_state, "PowerOff")
+    it "errors for a template" do
+      vm.update_attribute(:template, true)
 
-    expect(ws.root['vm'].power_state).to eq("off")
-    expect(ws.root['ae_result']).to eq("ok")
-  end
+      expect(vm.state).to eq("never")
+      expect { ws }.to raise_error(MiqAeException::ServiceNotFound)
+    end
+
+    it "retries for a vm in powered_on state" do
+      vm.update_attribute(:raw_power_state, "Running")
+
+      expect(ws.root['ae_result']).to eq("retry")
+      expect(ws.root['vm'].power_state).to eq("on")
+    end
 
-  it "errors for a template" do
-    @vm.update_attribute(:template, true)
-    expect(@vm.state).to eq("never")
+    it "returns 'ok' for a vm in unknown state" do
+      vm.update_attribute(:raw_power_state, "unknown")
 
-    expect { ws }.to raise_error(MiqAeException::ServiceNotFound)
+      expect(ws.root['vm'].power_state).to eq("unknown")
+      expect(ws.root['ae_result']).to eq("ok")
+    end
   end
 
-  it "retries for a vm in powered_on state" do
-    @vm.update_attribute(:raw_power_state, "Running")
+  context "Cloud" do
```

Comment:
@billfitzgerald0120 We have 4 tests for the Infrastructure method and 2 tests for the Cloud method. Can we add a few Cloud tests?